firmware/src/boards/cynthion_d11/usb_switch.c

```diff
@@ -0,0 +1,55 @@
+/**
+ * switch control for USB port shared by Apollo and FPGA
+ *
+ * This file is part of Apollo.
+ *
+ * Copyright (c) 2023 Great Scott Gadgets <info@greatscottgadgets.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "usb_switch.h"
+#include "apollo_board.h"
+#include "led.h"
+#include <hal/include/hal_gpio.h>
+
+/**
+ * Hand off shared USB port to FPGA.
+ */
+void hand_off_usb(void)
+{
+#if ((_BOARD_REVISION_MAJOR_ == 0) && (_BOARD_REVISION_MINOR_ < 6))
+	led_on(LED_D);
+#else
+	gpio_set_pin_level(USB_SWITCH, false);
+	gpio_set_pin_direction(USB_SWITCH, GPIO_DIRECTION_OUT);
+	led_off(LED_D);
+#endif
+}
+
+/**
+ * Take control of USB port from FPGA.
+ */
+void take_over_usb(void)
+{
+#if (((_BOARD_REVISION_MAJOR_ == 0) && (_BOARD_REVISION_MINOR_ >= 6)) || (_BOARD_REVISION_MAJOR_ == 1))
+	gpio_set_pin_level(USB_SWITCH, true);
+	gpio_set_pin_direction(USB_SWITCH, GPIO_DIRECTION_OUT);
+#endif
+	led_on(LED_D);
+}
+
+/**
+ * Handle switch control user request.
+ */
+void switch_control_task(void)
+{
+#if ((_BOARD_REVISION_MAJOR_ == 0) && (_BOARD_REVISION_MINOR_ < 6))
+	gpio_set_pin_pull_mode(PROGRAM_BUTTON, GPIO_PULL_UP);
+#else
+	gpio_set_pin_pull_mode(PROGRAM_BUTTON, GPIO_PULL_OFF);
+#endif
+	gpio_set_pin_direction(PROGRAM_BUTTON, GPIO_DIRECTION_IN);
```

Comment:
This is no longer necessary, the GPIO is configured here:
https://github.com/hathach/tinyusb/blob/5b08a65ad2e695332d56bd4676841a91f34dd5b1/hw/bsp/samd11/family.c#L80-L81